server/src/main/java/org/elasticsearch/readiness/ReadinessService.java

```diff
@@ -254,7 +254,7 @@ public void clusterChanged(ClusterChangedEvent event) {
     // protected to allow mock service to override
     protected boolean areFileSettingsApplied(ClusterState clusterState) {
         ReservedStateMetadata fileSettingsMetadata = clusterState.metadata().reservedStateMetadata().get(FileSettingsService.NAMESPACE);
-        return fileSettingsMetadata != null && fileSettingsMetadata.errorMetadata() == null;
+        return fileSettingsMetadata != null && fileSettingsMetadata.version() > 0;
```

Comment:
Unfortunately this isn't going to be reliable, version might be > 0 despite the first settings not being applied. 
The way I understand it, as soon as we're able to read / parse the file, we would write the version of it's metadata when writing the error state :/

Maybe we have to better differentiate errors that are permanent and transient. And in the latter case, allow to reprocess the file settings. Currently, if we see the same version again, we discard the update and not try again.